File: Practice Problems/nonConstructibleChange.py
def nonConstructibleChange(coins):
    # list must be sorted
    coins.sort()
    # Write your code here.
    currChange = 0
    for change in coins:
        if currChange + 1 < change:
            return currChange + 1
        else:
            currChange += change
            
    return currChange + 1

File: Practice Problems/nonConstructibleChange.cpp
#include <vector>
#include <algorithm>
using namespace std;

int nonConstructibleChange(vector<int> coins) {
  // Write your code here.
  vector<int>::iterator m = coins.begin();
  int change = 0;
  // empty condition
  
  sort(coins.begin(), coins.end());
  for(; m != coins.end(); m++) {
    //add each value if delta is greater than one we have our answer
      if (*m > change + 1) {
        return change + 1; 
      } else {
        change += *m;
      }
    }
    return change + 1;
  }

